### ISSUE TITLE
Privacy: Invalid HTML output with customization - "="-Character is added for empty HTML-attributes preceeding the src-attribute

### ISSUE THREAD

**Issue by strobocopter** [squidfunk/mkdocs-material#7625]
### Bug description

When adding a `<script>`-Tag via Customization (e.g. to conditionally load scripts depending on target-environment) the material/privacy plugin can produce invalid HTML-Code.

Example:
```
{% extends "base.html" %}

{% block extrahead %}
{% if config.extra.add_jquery %}
<script defer src="https://code.jquery.com/jquery-3.7.1.min.js"></script>
{% endif %}
{% endblock %}
```

This generates the following line in the output HTML-file:
```
<script defer= src="https://code.jquery.com/jquery-3.7.1.min.js"></script>
```

This seems to always happen if the `<script>`-Tag uses attributes without values (e.g. "defer" or "async") and if those attributes are placed before the "src"-attribute. In this case the plugin adds an equals-character to that attribute, but does not set a value, resulting in the browser to assume that "src= ...." is the value for the "defer"-attribute.

### Related links

- [Documentation for Privacy-Plugin](https://squidfunk.github.io/mkdocs-material/plugins/privacy/) -> not setting found to disable this behavior
- [Current Issues](https://github.com/squidfunk/mkdocs-material/issues) -> no issues found that look related

### Reproduction

[9.5.41-privacy-broken-html.zip](https://github.com/user-attachments/files/17394150/9.5.41-privacy-broken-html.zip)

### Steps to reproduce

See attached zip-archive...

1. Create empty mkdocs-material project
2. Add material/privacy plugin
3. Add customization HTML that adds a script-Tag
4. Have some attribute without value added to that script-Tag **_preceeding_** the src-attribute
5. Run mkdocs build and check the created html-Files

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [ ] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting! Now this is funny – I thought we fixed this, but it's actually of the category "how did this ever work"; as we're missing a `=` in the replacement (see fix in cc1508f1d). The problem stems from our use of a HTML4 parser, which is bundled with Python, and doesn't mandate us to install lxml, which is a huge dependency. For this to work, we must jump through some hoops, including this monkey patch for boolean attributes.

**Comment by squidfunk** [comment]
Released as part of [9.5.42](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.42).
